### ISSUE TITLE
[ENH] Probabilistic metric grid search integration

### ISSUE THREAD

**Issue by eenticott-shell** [alan-turing-institute/sktime#2233]
After developing probabilistic metrics in #2232 need to ensure they are compatible with useful features such as grid search for model parameters. There are two key problems that need to be solved for this:

1. proba metrics take in the output of `predict_quantile` or `predict_interval` (or `predict_proba`) where normal metrics just take predict. This means we need to change what predictions are used inside the grid search.

2. Some probabilistic metrics have their own hyperparameters. For example the quantile used in a pinball loss. Currently this is inferred from the data inputted, however for a grid search we will need to somehow tell it what quantile to produce. 

To solve 1. could either create some `set_default` function which determines what the forecaster implements for predict (_predict, _predict_quantile or _predict_interval) or use tags inside the grid search evaluation that retrieves the type of metric being used and calls the corresponding predict function.

To solve 2. we could do a small refactor to the probabilistic metrics, where we specify the hyperprameter(s) we want and it retrieves the correct data from the input (and raises an error if it isn't there). This will allow it to require a specific quantile but reduces flexibility as a user will have to instantiate a new metric class for each different set of quantiles they want to evaluate.

**Comment by fkiraly** [comment]
summarizing today's discussion, also looking at `evaluate`:

* I think we settled at giving all quantile losses an optional constructor argument `alpha`, and all interval losses an optional constructor argument `coverage`
* if the argument is not provided, all alpha/coverage passed (as part of the prediction) will be used to calculate the loss
* if the argument is provided, only the alpha/coverage passed will be used, and an error is raised if the specified alpha are not part of the prediction
* in `evaluate`, `alpha`/`coverage` can be passed via the loss, and the `evaluate` function gets the `alpha`/`coverage` from the loss and passes it to `predict_interval`/`predict_coverage`

Does this summarize it accurately?

The alternative design is indeed having `pred_args` in `evaluate`, though that may seem a bit odd to the user, since conceptually the `alpha` "belong" to the loss function, at least are very close to it.
